static/src/javascripts/projects/common/modules/ui/first-pv-consent-plus-engagement-banner.js

```diff
@@ -0,0 +1,153 @@
+    canShow as canShowEngagementBanner,
+} from 'common/modules/commercial/membership-engagement-banner';
+import {
+    track as trackFirstPvConsent,
+    bindClickHandlers as bindFirstPvConsentClickHandlers,
+    canShow as canShowFirstPvConsent,
+    messageCode as firstPvConsentMessageCode,
+    makeHtml as makeFirstPvConsentHtml,
+    template as firstPvConsentTemplate,
+    bindableClassNames as firstPvConsentBindableClassNames,
+} from 'common/modules/ui/first-pv-consent-banner';
+import marque36icon from 'svgs/icon/marque-36.svg';
+
+const messageCode: string = 'first-pv-consent-plus-engagement-banner';
+
+const bannerParams: EngagementBannerParams = engagementBannerParams(
+    geolocationGetSync()
+);
+
+const bannerTemplateParams: EngagementBannerTemplateParams = {
+    messageText: bannerParams.messageText,
+    ctaText: bannerParams.ctaText,
+    buttonCaption: bannerParams.buttonCaption,
+    linkUrl: addTrackingCodesToUrl({
+        base: bannerParams.linkUrl,
+        componentType: 'ACQUISITIONS_ENGAGEMENT_BANNER',
+        componentId: 'first_pv_consent_plus_support_the_guardian_banner',
+    }),
+};
+
+const bannerHtml = `
+    <div class="site-message js-site-message js-double-site-message site-message--banner site-message--double-banner" tabindex="-1" role="dialog" aria-label="welcome" aria-describedby="site-message__message" data-component="AcquisitionsEngagementBannerStylingTweaks_control">
+        <div class="js-engagement-banner-site-message site-message--engagement-banner">
+            <div class="gs-container">
+                <div class="site-message__inner js-site-message-inner">
+                    <div class="site-message__roundel">
+                        ${marque36icon.markup}
+                    </div>
+                    <div class="site-message__copy js-site-message-copy u-cf">
+                        ${acquisitionsBannerControlTemplate(
+                            bannerTemplateParams
+                        )}
+                    </div>
+                </div>
+            </div>
+        </div>
+    
+        <div class="js-first-pv-consent-site-message site-message--first-pv-consent" tabindex="-1" data-link-name="release message" role="dialog" aria-label="welcome" aria-describedby="site-message__message">
+            <div class="gs-container">
+                <div class="site-message__inner js-site-message-inner">
+                    <div class="site-message__roundel">
+                        ${marque36icon.markup}
+                    </div>
+                    <div class="site-message__copy js-site-message-copy u-cf">
+                        ${makeFirstPvConsentHtml(
+                            firstPvConsentTemplate,
+                            firstPvConsentBindableClassNames
+                        )}
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+`;
+
+class SubMessage extends Message {
```

Comment:
Since we have `SubMessage` as a concept I would prefer for it to perhaps do a bit more, Perhaps handle containing the banner htmls itself, maybe perhaps having a `SubMessageStacker(messages: SubMessage[]).show()` that would render them.

Reply:
Again, this is hard given how `message.js` works. Since right now we have no other use case for this `SubMessage`, I think we should consider how it can be generalised only if we need to render a different "double banner" before dotcomponents is fully live